src/state/routing/utils.ts

```diff
@@ -1,32 +1,50 @@
-import { MixedRouteSDK, Protocol } from '@uniswap/router-sdk'
+import { MixedRouteSDK } from '@uniswap/router-sdk'
 import { Currency, CurrencyAmount, Token, TradeType } from '@uniswap/sdk-core'
 import { Pair, Route as V2Route } from '@uniswap/v2-sdk'
 import { FeeAmount, Pool, Route as V3Route } from '@uniswap/v3-sdk'
+import { isPolygonChain } from 'constants/chains'
+import { nativeOnChain } from 'constants/tokens'
 
-import { GetQuoteResult, InterfaceTrade, V2PoolInRoute, V3PoolInRoute } from './types'
+import { GetQuoteArgs } from './slice'
+import {
+  ClassicTrade,
+  PoolType,
+  QuoteData,
+  QuoteState,
+  SwapRouterNativeAssets,
+  TradeResult,
+  V2PoolInRoute,
+  V3PoolInRoute,
+} from './types'
 
 /**
- * Transforms a Routing API quote into an array of routes that can be used to create
- * a `Trade`.
+ * Transforms a Routing API quote into an array of routes that can be used to
+ * create a `Trade`.
  */
 export function computeRoutes(
-  currencyIn: Currency | undefined,
-  currencyOut: Currency | undefined,
-  tradeType: TradeType,
-  quoteResult: Pick<GetQuoteResult, 'route'> | undefined
-) {
-  if (!quoteResult || !quoteResult.route || !currencyIn || !currencyOut) return undefined
+  tokenInIsNative: boolean,
+  tokenOutIsNative: boolean,
+  routes: QuoteData['route']
+):
+  | {
+      routev3: V3Route<Currency, Currency> | null
```

Comment:
nit: I think you could possibly infer those types / read them from Trade type, like suggested in the other comment. That way, don't have to redefine `routev3` `routev2` and `mixedRoute` types again. Makes it easier if we change the types we pass to generics.